mlflow/pyspark/ml/__init__.py

```diff
@@ -754,15 +754,35 @@ def log_post_training_metric(self, run_id, key, value):
 
 def _get_columns_with_unsupported_data_type(df):
     from mlflow.types.schema import DataType
+    from pyspark.ml.linalg import VectorUDT
 
     supported_spark_types = DataType.get_spark_types()
     unsupported_columns = []
     for field in df.schema.fields:
-        if field.dataType not in supported_spark_types:
+        if (field.dataType not in supported_spark_types) and not isinstance(field.dataType, VectorUDT):
             unsupported_columns.append(field)
     return unsupported_columns
 
 
+def _check_or_set_model_prediction_column(spark_model, input_spark_df):
+    from pyspark.ml import PipelineModel
+
+    prediction_column = "prediction"
+    if isinstance(spark_model, PipelineModel) and spark_model.stages[-1].hasParam("outputCol"):
+        from pyspark.sql import SparkSession
+
+        spark = SparkSession.builder.getOrCreate()
+        # do a transform with an empty input DataFrame
+        # to get the schema of the transformed DataFrame
+        transformed_df = spark_model.transform(spark.createDataFrame([], input_spark_df.schema))
```

Comment:
Is there a way for us to avoid the transformation by accessing the getter 
```python
spark_model.getPredictionCol()
```
to avoid spinning up an additional task pool? 
If I recall correctly, there are only a few edge cases (LDA, PCA, SQLTransformer, for instance) that don't have that attribute as part of the model object and instead use `getOutputCol` due to the nature of the models.

Reply:
I think using `spark_model.getPredictionCol()` is a breaking change.

Currently, `PyfuncModel.predict` assumes `prediction` column as the output column,

and user can set any column to "prediction" column name so that make mlflow `PyfuncModel.predict` recognize it as prediction column.

e.g., if user want to make the probability vector as the output column, user can set model param like:
```
LogisticRegression(maxIter=2) \
        .setPredictionCol("") \  # disable original prediction label output
        .setProbabilityCol("prediction")  # set probability vector outputs as prediction column
```

So `spark_model.getPredictionCol()` does not work in this case.

Reply:
Great point! While this workaround hack that users can do is something we should have supported similarly to how we support returning predict_proba in sklearn, this would definitely be a breaking change if we removed the support for this. 
Great explanation!